### PR TITLE
string class extension

### PR DESCRIPTION
**Pull request by omugeri**
peer review

### CODE REVIEW

File: app/string.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      return String.fromCharCode(upperChar.charCodeAt(0)+32);
+    });
+};
+String.prototype.ucFirst = function() {
+    return this.replace(/[a-z]/, (upperFirst) => {
+      return upperFirst.toUpper();
+    });
+};
+String.prototype.isQuestion = function() {
+    return /[?]/.test(this);
```
Comment (johnkariuki) [inline app/string.js:20]: Would `'?small babies cry.'` evaluate to `true` or `false`?

File: app/string.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+};
+String.prototype.words = function() {
+    var uWords = [];
+    var wordArray = this.replace(/\W+/g, ' ').trim().split(' ');
+    wordArray.forEach(function(word){
```
Comment (johnkariuki) [inline app/string.js:25]: Check out [filter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter). Would be nice to try it's implementation.

File: app/string.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      return String.fromCharCode(lowerChar.charCodeAt(0)-32);
+    });
+  };
+String.prototype.toLower = function() {
+    return this.replace(/[A-Z]/g, (upperChar) => {
+      return String.fromCharCode(upperChar.charCodeAt(0)+32);
+    });
+};
+String.prototype.ucFirst = function() {
+    return this.replace(/[a-z]/, (upperFirst) => {
```
Comment (kevgathuku) [inline app/string.js:15]: This does not work for cases where the first character is already uppercase. For example, running `'Kevin'.ucFirst()` returns `KEvin`

File: app/string.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+};
+
+String.prototype.wordCount = function() {
+    return this.words().length;
+};
+/*
+split the string from the decimal point then look for
+a pattern in the first element of the array.
+*/
+String.prototype.toCurrency = function() {
```
Comment (kevgathuku) [inline app/string.js:40]: Have you considered cases where the user doesn't provide a decimal.
e.g. `'462391729750'.toCurrency()` returns `"462,391,729,750."`
It has a hanging decimal point at the end.

**Pull request merged**